airflow/dag_processing/processor.py

```diff
@@ -16,610 +16,237 @@
-            raise
-        finally:
-            # We re-initialized the ORM within this Process above so we need to
-            # tear it down manually here
-            settings.dispose_orm()
-
-            result_channel.close()
-
-    def start(self) -> None:
-        """Launch the process and start processing the DAG."""
-        if conf.getboolean("scheduler", "parsing_pre_import_modules", fallback=True):
-            # Read the file to pre-import airflow modules used.
-            # This prevents them from being re-imported from zero in each "processing" process
-            # and saves CPU time and memory.
-            zip_file_paths = []
-            if zipfile.is_zipfile(self.file_path):
-                try:
-                    with zipfile.ZipFile(self.file_path) as z:
-                        zip_file_paths.extend(
-                            [
-                                os.path.join(self.file_path, info.filename)
-                                for info in z.infolist()
-                                if might_contain_dag(info.filename, True, z)
-                            ]
-                        )
-                except zipfile.BadZipFile as err:
-                    self.log.error("There was an err accessing %s, %s", self.file_path, err)
-            if zip_file_paths:
-                self.import_modules(zip_file_paths)
-            else:
-                self.import_modules(self.file_path)
-
-        context = self._get_multiprocessing_context()
-
-        pool_names = {p.pool for p in Pool.get_pools()}
-
-        _parent_channel, _child_channel = context.Pipe(duplex=False)
-        process = context.Process(
-            target=type(self)._run_file_processor,
-            args=(
-                _child_channel,
-                _parent_channel,
-                self.file_path,
-                f"DagFileProcessor{self._instance_id}",
-                self._dag_directory,
-                self._callback_requests,
-                pool_names,
-            ),
-            name=f"DagFileProcessor{self._instance_id}-Process",
+            log.exception("Failed to serialize DAG: %s", dag.fileloc)
+            dagbag_import_error_traceback_depth = conf.getint(
+                "core", "dagbag_import_error_traceback_depth", fallback=None
+            )
+            serialization_import_errors[dag.fileloc] = traceback.format_exc(
+                limit=-dagbag_import_error_traceback_depth
+            )
+    return serialized_dags, serialization_import_errors
+
+
+def _execute_callbacks(
+    dagbag: DagBag, callback_requests: list[CallbackRequest], log: FilteringBoundLogger
+) -> None:
+    for request in callback_requests:
+        log.debug("Processing Callback Request", request=request)
+        if isinstance(request, TaskCallbackRequest):
+            raise NotImplementedError("Haven't coded Task callback yet!")
+            # _execute_task_callbacks(dagbag, request)
```

Comment:
Worth adding a TODO too?

Reply:
I'll link to https://github.com/apache/airflow/issues/44354